prodigy/distribution/generating_function.py

```diff
@@ -31,6 +32,113 @@ def _term_generator(function: sympy.Poly):
+                return _sympy_symbol(prob_expr.var, rational, nonnegative,
+                                     **kwargs)
+            if isinstance(prob_expr, NatLitExpr):
+                return sympy.Integer(prob_expr.value)
+            if isinstance(prob_expr, RealLitExpr):
+                frac = prob_expr.to_fraction()
+                return sympy.Rational(frac.numerator, frac.denominator)
+            if isinstance(prob_expr, FunctionCallExpr):
+                if len(prob_expr.params[1]) != 0:
+                    raise ValueError(
+                        f'Cannot parse function with named parameters: {prob_expr}'
+                    )
+                return get_function(prob_expr.function)(
+                    *map(probably_to_sympy, prob_expr.params[0]))
+            if isinstance(prob_expr, BinopExpr):
+                op = prob_expr.operator
+                if op == Binop.DIVIDE:
+                    return probably_to_sympy(
+                        prob_expr.lhs) / probably_to_sympy(prob_expr.rhs)
+                if op == Binop.MINUS:
+                    return probably_to_sympy(
+                        prob_expr.lhs) - probably_to_sympy(prob_expr.rhs)
+                if op == Binop.TIMES:
+                    return probably_to_sympy(
+                        prob_expr.lhs) * probably_to_sympy(prob_expr.rhs)
+                if op == Binop.POWER:
+                    return probably_to_sympy(prob_expr.lhs)**probably_to_sympy(
+                        prob_expr.rhs)
+                if op == Binop.MODULO:
+                    return probably_to_sympy(
+                        prob_expr.lhs) % probably_to_sympy(prob_expr.rhs)
+                if op == Binop.PLUS:
+                    return probably_to_sympy(
+                        prob_expr.lhs) + probably_to_sympy(prob_expr.rhs)
+                raise ValueError(f'Unsupported operand: {op}')
+            raise ValueError(f'Unsupported expression type: {prob_expr}')
+
+        s = probably_to_sympy(expr)
+
+    else:
+        s = sympy.parse_expr(
+            str(expr),
+            transformations=sympy.parsing.sympy_parser.standard_transformations
+            + (sympy.parsing.sympy_parser.convert_xor,
+               sympy.parsing.sympy_parser.rationalize),
+            global_dict={
+                'Symbol':
+                lambda x: _sympy_symbol(x, rational, nonnegative, **kwargs),
+                'Function':
+                get_function,
+                'Integer':
+                sympy.Integer,
+                'Float':
+                sympy.Float,
+                'Rational':
+                sympy.Rational
+            })
+
+    return s
+
+
+def _sympy_symbol(name: Any,
+                  rational=True,
+                  nonnegative=True,
+                  **kwargs) -> sympy.Symbol:
+    """
+    Creates a rational and nonnegative sympy Symbol.
+    
+    Note that `sympy.Symbol('x') == sympy.Symbol('x', rational=True, nonnegative=True)`
```

Comment:
better say "evaluates to ``false``". Also a sentence describing that you are not referring to the calling convention but to the fact that comparing two variables with the same name but different domains simply yields ``false`` helps understanding that.